tecs/tecs.cpp

```diff
@@ -473,8 +475,9 @@ void TECS::_update_pitch_setpoint()
 					max((_pitch_setpoint_min - _pitch_setpoint_unc) * climb_angle_to_SEB_rate / _pitch_time_constant, 0.0f));
 	}
 
-	// Update the pitch integrator state
+	// Update the pitch integrator state. If gain is zero, force state to zero, too, to avoid that it stays at large constant values
 	_pitch_integ_state = _pitch_integ_state + pitch_integ_input * _dt;
+	if(_integrator_gain < 0.001f) _pitch_integ_state = 0.0f;
```

Comment:
Same as above. If you already know you're going to zero something you might as well save the effort of calculating it.

``` C++
if (_integrator_gain > 0.0f) {
    _pitch_integ_state = _pitch_integ_state + pitch_integ_input * _dt;
} else {
    _pitch_integ_state = 0.0f;
}
```